src/ferui-components/tree-view/tree-view-component.ts

```diff
@@ -502,4 +537,28 @@ export class FuiTreeViewComponent<T> implements OnInit, OnDestroy {
       });
     }
   }
+
+  /**
+   * Check if configuration width is a fixed pixel length, if not set a Dom Observer to adjust Tree View virtual scroller
+   * width based on its container when any container size happens (including window resize)
+   */
+  private domObserverHandler(): void {
+    if (!this.treeViewStyles.width.includes('px')) {
+      this.domObserver = DomObserver.observe(this.el.nativeElement.firstChild, undefined, () => {
+        this.originalWidth = this.el.nativeElement.firstElementChild.clientWidth;
+        const biggestNode = this.scrollViewArray.reduce((acc, current) => {
+          return current.width ? Math.max(acc, current.width) : acc;
+        }, 0);
+        this.treeViewUtils.defaultScrollerWidth = Math.max(biggestNode, this.originalWidth);
+        let padding = 0;
+        if (this.originalWidth > biggestNode) {
+          // if container width is bigger than nodes width we will subtract to adjust scroller, default 20px
+          const diff = Math.abs(this.originalWidth - biggestNode);
+          padding = diff > 20 ? 20 : diff;
```

Comment:
Here is the magical 20px value. You should use a constant and share it between your files.

Reply:
Done